### PR TITLE
Routing to Controller Actions PR

### PR DESCRIPTION
**Pull request by ssargent**
Change is to mvc/controllers/routing.rst Routing to Controller Actions.  Also includes a sample project that I created to test the routes, functionality etc..  Happy to make any requested changes.  Let me know what should be expanded upon, what should be condensed etc..

### CODE REVIEW

File: common/authors.txt [hunk 1]
```diff
@@ -7,4 +7,5 @@
 .. _Erik Reitan: https://github.com/Erikre
 .. _Pranav Rastogi: https://github.com/rustd
 .. _Tom Archer: https://github.com/tomarcher
+.. _Scott Sargent:https://github.com/ssargent
```
Comment (danroth27) [inline common/authors.txt]: Add a space after the colon to separate your name from the link

**Comment by danroth27** [comment]
@rynowak Please review

**Comment by danroth27** [comment]
Thanks for the contribution! This looks like a good initial article on attribute routing (convention based routing could be covered separately). The article could use a simple intro. Also I think we need to go beyond illustrative examples in the some of the sections to add some more reference content, like what inline constraints are available, what route tokens are available, syntax for optionality and default values, etc. I think we need to do an editorial pass on the content to make the voice consistent with the rest of our docs and also to clean up the RST. @rachelappel Is this something you could help out with?

I see you put the sample code in the common folder. We only use the common folder for samples that are frequently used across multiple projects. Right now the only samples that we have in common are the latest project templates from Visual Studio. All other sample code should live in a companion folder for the article. You can take a look at other articles in the repo to see how this is typically setup.

Ideally any sample code included gets directly referenced by the article using the [Sphinx literalinclude](http://sphinx-doc.org/markup/code.html#directive-literalinclude) directive instead of inline codeblocks. This makes it easier to make sure the code snippets in the docs all compile. We do try to keep the samples as simple and illustrative as possible as this makes the code easier to maintain over time.

Looks like the sample code was based on beta5. We should check to make sure that the code still works as expected on rc1.

This PR also contains a new top level Routing node in the MVC TOC with a bunch of sub directories and empty files. I think this node can be removed for now since it doesn't contain any content and I'm not really sure it's needed.

Let us know how much of this you want to take on yourself or if you prefer that we handle the additional required updates. Thank you again for helping to document ASP.NET 5!

**Comment by rachelappel** [comment]
@danroth27 Happy to! I'll hold off until @ssargent  lets us know where we can best help.

**Comment by ssargent** [comment]
Sorry for the delay.  I'll take an initial pass at this and have it in by monday morning.  I figure I'll tackle the low hanging fruit, formatting, intro, removing unneeded things ie TOC, include vs inline.  I'll definitely need help with the voice, I'll check some other content to get an idea but I've a feeling I'm not going to stick the landing on that so-to-speak.

**Comment by Rick-Anderson** [comment]
moved to #1374

**Pull request closed without merging**